tests/test_databases.py

```diff
@@ -514,3 +514,95 @@ def run_sync(*args, **kwargs):
         test_complete.set()
         await task_1
         await task_2
+
+
+@pytest.mark.parametrize("database_url", DATABASE_URLS)
+@async_adapter
+async def test_connection_context_with_raw_connection(database_url):
+    """
+    Test connection contexts with respect to the raw connection.
+    """
+    async with Database(database_url) as database:
+        async with database.connection() as connection_1:
+            async with database.connection() as connection_2:
+                assert connection_1 is connection_2
+
+                raw_connection_0 = await database.raw_connection()
+                raw_connection_1 = await connection_1.raw_connection()
+                raw_connection_2 = await connection_2.raw_connection()
+
+                assert raw_connection_0 is raw_connection_1 is raw_connection_2
+                assert raw_connection_0 is connection_1._connection._connection
```

Comment:
Using properties I think we'd really want to change this to something like this...

```python
    async with Database(database_url) as database:
        async with database.connection() as connection_1:
            async with database.connection() as connection_2:
                assert connection_1 is connection_2
                assert connection_1.raw_connection is connection_2.raw_connection
```

Reply:
yes, that would be neat